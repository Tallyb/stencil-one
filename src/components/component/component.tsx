import { Component, Prop, h, JSX, Event, State, EventEmitter, Method } from '@stencil/core';
import { format } from '../../utils/utils';

@Component({
  tag: 'my-component',
  styleUrl: 'component.css',
  shadow: true
})
export class MyComponent {
  /**
   * The first name
   */
  @Prop() first: string;

  /**
   * The middle name
   */
  @Prop() middle: string;

  /**
   * The last name
   */
  @Prop() last: string;

  @State() buttonFace: string = 'Click Me!';

  @Event() buttonClicked: EventEmitter<string>;
  
  @Method() 
  async updateFace(value: string): Promise<string>{
    this.buttonFace = value;
    return this.buttonFace.toUpperCase();
  }

  getOne (): string {
    return 'This is a string'
  }
  private getText(): string {
    return format(this.first, this.middle, this.last);
  }
    
  onClicked(value: string) {
     console.log('Value Clicked is ', value);
  }

  render() : JSX.Element {
    return (
      <div class="nice">
      <span>
        Hello, World! I'm {this.getText()}
      </span>
      <button onClick={()=>this.onClicked('ABCD')}>
        {this.buttonFace}
      </button>
      </div>
    );
  }
}
