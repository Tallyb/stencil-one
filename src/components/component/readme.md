# my-component



<!-- Auto Generated Below -->


## Properties

| Property | Attribute | Description     | Type     | Default |
| -------- | --------- | --------------- | -------- | ------- |
| `first`  | `first`   | The first name  | `string` | `''`    |
| `last`   | `last`    | The last name   | `string` | `''`    |
| `middle` | `middle`  | The middle name | `string` | `''`    |


## Events

| Event           | Description | Type                  |
| --------------- | ----------- | --------------------- |
| `buttonClicked` |             | `CustomEvent<string>` |


## Methods

### `updateFace(value: string) => Promise<string>`



#### Returns

Type: `Promise<string>`




----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
