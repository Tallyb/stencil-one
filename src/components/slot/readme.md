# my-complex-prop



<!-- Auto Generated Below -->


## Properties

| Property | Attribute | Description | Type       | Default     |
| -------- | --------- | ----------- | ---------- | ----------- |
| `data`   | --        |             | `string[]` | `undefined` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
