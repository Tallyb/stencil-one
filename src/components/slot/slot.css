.cool {
  display: flex;
  flex-direction: column;
  height: 3rem;
  border-color: green;
}
h {
  color: cyan;
}