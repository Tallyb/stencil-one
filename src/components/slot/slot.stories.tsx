/** @jsx h */
/** @jsxRuntime classic */

import {h} from '@stencil/core';


export default {
  title: 'My slot',
};

export const Default = {
  render: () => 
    <my-slot>
      <my-component first={'Juliaa'} last={'Roberts'}></my-component>
    </my-slot>
};

