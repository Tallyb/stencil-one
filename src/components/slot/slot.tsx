import { Component, Prop, h, JSX} from '@stencil/core';

@Component({
  tag: 'my-slot',
  styleUrl: 'slot.css',
  shadow: true
})
export class MySlotComponent {

  @Prop() values: Array<string> = [];

  render() : JSX.Element {
    return (
      <div class="cool">
        <h>I am a header </h>
        <div>
          <slot/>
        </div>
        <h>I am a footer </h>
      </div>
    );
  }
}
