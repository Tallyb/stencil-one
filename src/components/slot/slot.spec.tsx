import { h } from '@stencil/core';
import { newSpecPage } from '@stencil/core/testing';

import { MySlotComponent } from './slot';
import { MyComponent } from '../component/component';

describe('complex prop', () => {
  it('should render', async () => {
    const page = await newSpecPage({
      components: [MySlotComponent, MyComponent],
      template: () => (
        <my-slot>
          <my-component first={'Julia'} last={'Roberts'}></my-component>
        </my-slot>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });
});
