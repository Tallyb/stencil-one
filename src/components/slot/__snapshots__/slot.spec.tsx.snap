// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`complex prop should render 1`] = `
<my-slot>
  <mock:shadow-root>
    <div class="cool">
      <h>
        I am a header
      </h>
      <div>
        <slot></slot>
      </div>
      <h>
        I am a footer
      </h>
    </div>
  </mock:shadow-root>
  <my-component>
    <mock:shadow-root>
      <div class="nice">
        <span>
          Hello, World! I'm Julia Roberts
        </span>
        <button>
          Click Me!
        </button>
      </div>
    </mock:shadow-root>
  </my-component>
</my-slot>
`;
